projects/ngx-formentry/src/form-entry/form-renderer/form-renderer.component.html

```diff
@@ -290,11 +290,11 @@
             [min]="node.question.extras.questionOptions.min"
             [max]="node.question.extras.questionOptions.max"
             [formControlName]="node.question.key"
-            [attr.placeholder]="node.question.placeholder"
+            [attr.placeholder]="node.question.placeholder | translate"
           >
           </ofe-number-input>
 
-          <input
+          <ofe-number-input
```

Comment:
This seems like a mistake no?

Reply:
I think that is the correct one because it was throwing an error as input just so I used the same implementation we did on the form `ofe-form` and did the same for input and it stopped showing the error.

Reply:
So, `ofe-number-input` is a custom tag for the custom element [defined here](https://github.com/openmrs/openmrs-ngx-formentry/tree/main/projects/ngx-formentry/src/components/number-input). I think this exists largely so we can match the [carbon styling](https://angular.carbondesignsystem.com/?path=%2Fstory%2Fcomponents-number--basic) for numeric inputs. So we sue it above when a numeric input is being rendered (note the `*ngSwitchCase="'number'"`). This component, however, is a simple text input which shouldn't need additional styling and was just using the standard HTML `input` tag. If there were errors, we should figure why those errors occurred and fix them, because the numeric input styling is likely to look wrong for text inputs

Reply:
Okay, I will look into that.

Reply:
`theme` seems to be causing the issue.

Reply:
I have seen [this](https://github.com/openmrs/openmrs-ngx-formentry/blob/099f32836527142399e2469347e80b85c1aa33ea/projects/ngx-formentry/src/form-entry/form-renderer/form-renderer.component.html#L286) in the main branch and I have seen another solution you mention for adding carbon design class, it is confusing when we should use `ofe-number-input` and when we need to use just `input`.

Reply:
The guiding rule here is that this section of the form-renderer.component.html is, in essence, a bit switch statement that determines the tag based on the question, so in this context, each tag should only appear once.

In your screenshot, but not in the PR, I see you're missing the `ofeTextInput` that was there originally which is why you're seeing the error. `ofeTextInput` is the directive that adds the necessary styling to this element, including the `theme` attribute. If you just add that back in, the error should go away.